src/compiler/check/writer.ts

```diff
@@ -0,0 +1,761 @@
+/// <reference path="../binder.ts" />
+
+/* @internal */
+namespace ts {
+    export function createWriter(
```

Comment:
I thought about it and I actually like 'writer'. Nothing else in our code actually writes things except the TextWriter, which is small and focussed. This file handles writing everything else.